### ISSUE TITLE
[css3-positioning] position:sticky should create a stacking context

### ISSUE THREAD

**Issue by smfr** [w3c/csswg-drafts#1053]
I think position:sticky should create a stacking context, as we have resolved to do for position:fixed.

**Comment by asjf2** [comment]
+1

I've hit this in practice today and have no workaround

**Comment by keithjgrant** [comment]
> as we have resolved to do for position:fixed

Does this mean position: fixed will start creating a stacking context even without the use of z-index? Is that in the current working draft?

**Comment by Tyler-H** [comment]
@keithjgrant The working group resolved to have position:fixed create a stacking context in their 2015-07-01 meeting. While it was left out subsequent spec revisions on accident, it was [added in](https://github.com/w3c/csswg-drafts/commit/4aaf53107a4b370310104994e64b6aec2eaf65a9) on November 18th of 2016.

See https://github.com/w3c/csswg-drafts/issues/451#issuecomment-261457816 for more info.

tl;dr Yes, position:fixed already creates a stacking context even without the use of z-index.

**Comment by SelenIT** [comment]
I believe that @tabatkins's reasoning [in the related issue](2326#issuecomment-366337621) and the fact that Edge managed to implement `position:sticky` without necessarily making it stacking context are good argument _against_ making this change in the spec. The `position:fixed` special case was introduced rather ad hoc, mostly to match the already long existed implementation that was too hard to "fix", but I hope the `position:sticky` case is different and we can do better this time.

@smfr, could you please clarify what is the most difficult problem of implementing `position:sticky` without making it stacking context for `z-index: auto`?

**Comment by jonjohnjohnson** [comment]
@smfr Any new thoughts on this before adoption makes it too late and interop causes standards to resolve on a possibly subpar decision? Especially since @tabatkins and @FremyCompany weighed in on current language sufficing? Even more important while https://github.com/w3c/csswg-drafts/issues/2717 is consolidating how/if stacking contexts automatically sort in z space?

**Comment by chrishtr** [comment]
Chromium already treats both position: fixed and position: sticky as stacking contexts.
See [here](https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/style/computed_style.cc?q=computed_style.cc&sq=package:chromium&dr&l=932).

The reason that such elements are stacking contexts is that they need to move with respect to scroll, and in particular composited scrolling layers. Therefore they are put in their own composited layer, and
if that layer were *not* a stacking context, then a lot of unnecessary implementation complexity
would result due to the contents of the fixed/sticky subtree not being atomically rendered.

**Comment by jonjohnjohnson** [comment]
@chrishtr I think Edge is now the only implementation that doesn't treat sticky as a stacking context. And implementation complexity regarding layers is completely understandable, I think we all realized this was the case, but I wish standards would have found agreement on this before implementations were shipped.

Should an issue be filed with Edge for interop?
Or is that premature and we should wait for this to be added to the groups agenda on the off chance this issues suggestion isn't taken?

**Comment by chrishtr** [comment]
I think the next step is to file an issue with edge and create wpt tests. And edit the spec to agree with behavior on the other browsers.

**Comment by chrishtr** [comment]
Edge no longer exists as a different browser. The spec needs to be edited to make position:sticky a stacking context and a WPT test added. @tabatkins

**Comment by jcubic** [comment]
Was this resolved? Does `position: sticky` create stacking context in other browsers? I found an issue with `position: sticky` on a table. That may be related to a problem that `position: sticky` doesn't create stacking context. The issue is visible in Firefox and Chrome.

**Comment by tabatkins** [comment]
Agenda+ to confirm that stickypos should create a stacking context just like fixpos does, for the reasoning given by @chrishtr in <https://github.com/w3c/csswg-drafts/issues/1053#issuecomment-429511668>

**Comment by fantasai** [comment]
We should remember to republish after the CSSWG resolution here, it's the only significant thing blocking republication.
